casper/src/main/scala/coop/rchain/casper/util/ProtoUtil.scala

```diff
@@ -307,19 +307,12 @@ object ProtoUtil {
       extraBytes = ByteString.EMPTY
     )
 
-    val hash = hashUnsignedBlock(block)
+    val hash = hashBlock(block)
 
     block.copy(blockHash = hash)
   }
 
-  def hashUnsignedBlock(blockMessage: BlockMessage): BlockHash = {
```

Comment:
@tgrospic @nzpr I made the decision by myself to remove this method forever but I am open to see if you guys get any other ideas. I don't think we should keep two hashing methods for the same `BlockMessage` object.

Reply:
I didn't actually notice here. We didn't actually validate genesis blockHash before.

Reply:
```scala
 private def validateBlock(block: BlockMessage): F[Boolean] = { 
     Validate.blockHash(block).map(_ == Right(Valid)) 
 } 
```
like this?

I am not sure whether we should change this or not. I got an ultimate question for this. Should we maintain the old network for user to replay and validate?

Reply:
Yes, that is the fix, now we can validate all block in the same way.

> We didn't actually validate genesis blockHash before.

Yes, but LFS on main net was not downloading genesis block anyway.

> I am not sure whether we should change this or not. I got an ultimate question for this. Should we maintain the old network for user to replay and validate?

Yes, but to work with history networks older versions of node should be used. Similar how latest Ethereum node cannot process all previous blocks.

Reply:
> Yes, that is the fix, now we can validate all block in the same way.

The exist genesis block https://revdefine.io/#/blocks/986addc3dfa12b179eaa40e38d77aec3da0530b9ca2243271436a135055229dd/ can not be validated this way.

Reply:
Do you see situation where this can be a problem? We have v0.10.2 which can.

We can include this PR with HF1 release?

Reply:
PR is now rebased on `feature/hard-fork-1`.